src/WebJobs.Extensions.DurableTask/Listener/TaskHttpActivityShim.cs

```diff
@@ -42,32 +42,40 @@ public async override Task<string> RunAsync(TaskContext context, string rawInput
             HttpRequestMessage requestMessage = await this.ConvertToHttpRequestMessage(durableHttpRequest);
 
             HttpResponseMessage response;
-            if (durableHttpRequest.Timeout == null)
+            try
             {
-                response = await this.httpClient.SendAsync(requestMessage);
-            }
-            else
-            {
-                try
+                if (durableHttpRequest.Timeout == null)
                 {
-                    using (CancellationTokenSource cts = new CancellationTokenSource())
-                    {
-                        cts.CancelAfter(durableHttpRequest.Timeout.Value);
-                        response = await this.httpClient.SendAsync(requestMessage, cts.Token);
-                    }
+                    response = await this.httpClient.SendAsync(requestMessage);
                 }
-                catch (OperationCanceledException ex)
+                else
                 {
-                    TimeoutException e = new TimeoutException(ex.Message + $" Reached user specified timeout: {durableHttpRequest.Timeout.Value}.");
+                    try
+                    {
+                        using (CancellationTokenSource cts = new CancellationTokenSource())
+                        {
+                            cts.CancelAfter(durableHttpRequest.Timeout.Value);
+                            response = await this.httpClient.SendAsync(requestMessage, cts.Token);
+                        }
+                    }
+                    catch (OperationCanceledException ex)
+                    {
+                        TimeoutException e = new TimeoutException(ex.Message + $" Reached user specified timeout: {durableHttpRequest.Timeout.Value}.");
 
-                    string details = Utils.SerializeCause(e, this.config.ErrorDataConverter);
-                    throw new TaskFailureException(e.Message, e, details);
+                        string details = Utils.SerializeCause(e, this.config.ErrorDataConverter);
+                        throw new TaskFailureException(e.Message, e, details);
+                    }
                 }
-            }
 
-            DurableHttpResponse durableHttpResponse = await DurableHttpResponse.CreateDurableHttpResponseWithHttpResponseMessage(response);
+                DurableHttpResponse durableHttpResponse = await DurableHttpResponse.CreateDurableHttpResponseWithHttpResponseMessage(response);
 
-            return JsonConvert.SerializeObject(durableHttpResponse);
+                return JsonConvert.SerializeObject(durableHttpResponse);
+            }
+            catch (HttpRequestException ex)
+            {
+                string details = Utils.SerializeCause(ex, this.config.ErrorDataConverter);
```

Comment:
You mentioned in the description that `HttpRequestException` doesn't serialize correctly. How is it that we don't run into the same problem here?

Reply:
The details on line 76 are correctly serialized with the right information, but the `TaskFailureException` that's created doesn't serialize the `Message` property for `ex`, the InnerException.

For example, on line 77, `ex.Message` is "No such host is known.", but then in `StartNewOrchestration` in `DurableOrchestrationContext`, the `e.InnerException.Message` turns out to be "'System.Net.Http.HttpRequestException' was thrown."

Reply:
Should we track a fix to that serialization problem?